pythonforandroid/build.py

```diff
@@ -484,7 +484,9 @@ def set_archs(self, arch_names):
             ', '.join([arch.arch for arch in self.archs])))
 
     def prepare_bootstrap(self, bs):
-        bs.ctx = self
+        if bs:
+            # XXX: this condition fixes tox ci tests, dont know when in reality we have no bootstrap
```

Comment:
It sounds like this is because prepare_bootstrap is being passed None when it shouldn't be. I would prefer to keep the code with prepare_bootstrap expecting a real argument, and the tests passing one if necessary, but I haven't looked at the details of the failure - do you think there's some reason that wouldn't be practical?

Reply:
I agree, thats the reason why I put in that comment. prepare_bootstrap now raises a TypeError when bootstrap is None, fixed tests to reflect that